api/categorisation.go

```diff
@@ -25,26 +25,33 @@ func (api *API) RetrieveDefaultCategorisation(dimension *model.Dimension, datase
 	}
 
 	names := make([]string, 0)
-	for _, edge := range cats.Dataset.Variables.Edges {
+
+	for edgeIndex := range cats.Dataset.Variables.Edges {
+		edge := cats.Dataset.Variables.Edges[edgeIndex]
+
 		if len(edge.Node.MapFrom) > 0 {
-			for _, mapFrom := range edge.Node.MapFrom {
-				for _, mappedSource := range mapFrom.Edges {
-					for _, mappedSourceEdge := range mappedSource.Node.IsSourceOf.Edges {
+			for mapFromIndex := range edge.Node.MapFrom {
```

Comment:
range provides both index and element, unless I am missing something ??

Reply:
https://go-critic.com/overview.html#rangevalcopy <-- rangeValCopy issue, although it looks like I should've used &pointer as well.